overlord/snapstate/handlers.go

```diff
@@ -324,19 +324,28 @@ func (m *SnapManager) installOneBaseOrRequired(t *state.Task, snapName string, c
 		return nil, err
 	}
 
-	inProgress := func(snapName string) (bool, error) {
+	shouldAttemptInstall := func(snapName string) (bool, error) {
 		linkTask, err := findLinkSnapTaskForSnap(st, snapName)
 		if err != nil {
 			return false, err
 		}
 
 		if linkTask == nil {
 			// snap is not being installed
+			return true, nil
+		}
+
+		// if we are remodeling, then we should return early due to the way that
+		// tasks are ordered by the remodeling code. specifically, all snap
+		// downloads during a remodel happen prior to snap installation. thus,
+		// we cannot wait for snaps to be installed here. see remodelTasks for
+		// more information on how the tasks are ordered.
+		if deviceCtx.ForRemodeling() {
 			return false, nil
 		}
 
-		// snap is being installed, retry later
-		return true, nil
+		// snap is either being installed or updated
+		return false, onInFlight
```

Comment:
I'm not sure that return directly onInFlight is helpful here

Reply:
I return `onInFlight` here to distinguish between when we are remodeling or when we aren't.

Reply:
I reworked how this looks in the most recent commit. Since we don't actually want to go into `updatePrereqIfOutdated` during a remodel since it tries to reach out to the store (cc @MiguelPires to make sure I'm not lying), I think we can just move the check for remodeling earlier in the function.